homeassistant/components/vesync/config_flow.py

```diff
@@ -57,3 +58,35 @@ async def async_step_user(
             title=username,
             data={CONF_USERNAME: username, CONF_PASSWORD: password},
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle re-authentication with vesync."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm re-authentication with vesync."""
+
+        if user_input:
+            reauth_entry = self._get_reauth_entry()
+            username = user_input[CONF_USERNAME]
+            password = user_input[CONF_PASSWORD]
+            data = {
+                **reauth_entry.data,
+                CONF_USERNAME: username,
+                CONF_PASSWORD: password,
+            }
+
+            manager = VeSync(username, password)
+            login = await self.hass.async_add_executor_job(manager.login)
+            if login:
+                return self.async_update_reload_and_abort(reauth_entry, data=data)
```

Comment:
If we can't login, I think we should show an error. We should also take in account any exceptions raised because we could not connect to VeSync (for whatever reason)

Reply:
I think the usual pattern to raise `ConfigEntryAuthFailed` which causes HomeAssistant to present authentication flow.

Reply:
In this section right now if login failure it displays the login box again.   It doesn't state why though.  If we decide to proceed I am happy to put the work in to get this display another error that the new creds also failed. 
`ConfigEntryAuthFailed` is thrown within the init login to trigger this flow.

Reply:
I have corrected this so it shows an error on re-auth failure.  That is a few lines lower - it is uploading now.